src/fasta-recs.c

```diff
@@ -9,8 +40,63 @@ int main(int argc, char const *argv[])
     }
 
     const char *fasta_fname = argv[1];
-    printf("I should be parsing %s now, but someone forgot to implement that.\n",
-           fasta_fname);
+
+    char *content = readFile(fasta_fname);
+
+    //Parsing into DNA chains
+    int pos = 0;
+    const int length = strlen(content);
+
+    // Find every chain
+    while(pos<length){
```

Comment:
If content is zero-terminated, then you can check for zero to find out if you are at the end.

```c
for (char *p = content; *p; p++)
{
    // do stuff with the character *p
}
```

The test `*p` looks at the character that `p` points to, to see if it is the end of the string. That is the usual way this is done in C.